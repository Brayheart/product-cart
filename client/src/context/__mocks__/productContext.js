import React from 'react';
export const ProductContext = React.createContext({
    product: [],
})

export const ProductContextProvider = ({ children }) => {
    return (
        <ProductContext.Provider value={[
            {
                imgUrl: 'https://www.mobygames.com/images/covers/l/153013-pokemon-red-version-game-boy-front-cover.jpg',
                product: 'DMG-P-USA',
                productName: 'Pokemon Red',
                price: 51.5,
                quantity: 1
            }
        ]}>
            {children}
        </ProductContext.Provider>
    )
};