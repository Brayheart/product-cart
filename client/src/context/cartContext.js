import React, {useEffect, useState} from 'react';
import { getStorageCart } from '../utils/localStorage';

export const CartContext = React.createContext({
    cart: [],
    total: 0,
    updateCart: () => {},
})

export const CartContextProvider = (props) => {

    const updateCart = (cartData) => setState({...state, ...cartData})
    
    const initialStore = {
        cart: [],
        total: 0,
        updateCart: updateCart
    }

    const [state, setState] = useState(initialStore)

    useEffect(() => {
        const {cart, total} = getStorageCart();
        setState({...state, cart, total})
    }, [])
    
    return (
        <CartContext.Provider value={state}>
            {props.children}
        </CartContext.Provider>
    )
}