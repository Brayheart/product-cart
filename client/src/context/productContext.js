import React, { useEffect, useState } from 'react';


export const ProductContext = React.createContext({
    product: [],
})

export const ProductContextProvider = (props) => {

    const [state, setState] = useState(null)

    useEffect(() => {
        const fetchAllProducts = async () => {
            try {
                let products = await fetch('http://localhost:8080/api/products/all').then((response) => response.json())
                setState(products)
            } catch (error) {
                setState([])
            }
        }

        fetchAllProducts()
    }, [])
    
    return (
        <ProductContext.Provider value={state}>
            {props.children}
        </ProductContext.Provider>
    )
}