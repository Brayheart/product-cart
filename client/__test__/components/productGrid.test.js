import React, { useContext } from 'react'
import renderer from 'react-test-renderer';
import ProductGrid from '../../src/components/productGrid';
import { CartContextProvider } from '../../src/context/cartContext';
import { ProductContextProvider } from '../../src/context/productContext';
jest.mock('./../../src/context/productContext');

beforeEach(() => {
  // Clear all instances and calls to constructor and all methods:
});

it('displays a grid of products', () => {

    const component = renderer.create(
        <ProductContextProvider>
            <CartContextProvider>
                <ProductGrid />
            </CartContextProvider>
        </ProductContextProvider>
    )

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot()
})