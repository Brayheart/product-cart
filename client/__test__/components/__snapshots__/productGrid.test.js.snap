// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a grid of products 1`] = `
<section
  className="product-grid-container"
>
  <article
    className="product-card-container"
    data-cart-index="0"
  >
    <div
      className="product-card-inner"
    >
      <div
        className="product-card-image-container"
      >
        <div
          className="product-card-image-inner"
        >
          <img
            src="https://www.mobygames.com/images/covers/l/153013-pokemon-red-version-game-boy-front-cover.jpg"
          />
        </div>
      </div>
      <div
        className="product-card-details"
      >
        <div
          className="product-card-details-name"
        >
          <h3>
            Pokemon Red
          </h3>
        </div>
        <div
          className="product-card-details-price"
        >
          $
          51.50
        </div>
        <div
          className="product-card-details-cart-count"
        >
          Cart Count: 
          0
        </div>
        <div
          className="product-card-details-btns"
        >
          <button
            className="product-card-detail-btn btn-add"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="product-card-detail-btn btn-remove"
            onClick={[Function]}
          >
            Remove
          </button>
        </div>
      </div>
    </div>
  </article>
</section>
`;
